assets/src/components/detourListPage.tsx

```diff
@@ -29,19 +60,35 @@ export const DetourListPage = () => {
       )}
       {detours && (
         <>
-          {detours.active && <DetoursTable data={detours.active} />}
-          {detours.draft && <DetoursTable data={detours.draft} />}
-          {detours.past && <DetoursTable data={detours.past} />}
+          {detours.active && (
+            <DetoursTable data={detours.active} onOpenDetour={onOpenDetour} />
+          )}
+          {detours.draft && (
+            <DetoursTable data={detours.draft} onOpenDetour={onOpenDetour} />
+          )}
+          {detours.past && (
+            <DetoursTable data={detours.past} onOpenDetour={onOpenDetour} />
+          )}
         </>
       )}
 
-      {showDetourModal && (
+      {/* Need to be separate modals to force the modal's state machine to refresh */}
+      {showDetourModal && !detourId && (
         <DetourModal
           onClose={() => setShowDetourModal(false)}
           show
           originalRoute={{}}
         />
       )}
+
+      {showDetourModal && stateOfDetourModal && (
+        <DetourModal
+          onClose={onCloseDetour}
+          show
+          originalRoute={{}}
+          {...(stateOfDetourModal ? { snapshot: stateOfDetourModal } : {})}
```

Comment:
```suggestion
          snapshot={stateOfDetourModal}
```

Reply:
Oh yep, relic of trying to combine into one modal previously